### PR TITLE
Add Dark Mode

### PR DESCRIPTION
**Pull request by danhp**
This build upon what was done in #26 but with a few enhancements
Those being:
- Themed it to match the sister app Caprine.
- Bring in the auto dark feature that we previously implemented.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Nice!

// @paulmolluzzo

File: index.js [hunk 1]
```diff
@@ -83,6 +83,7 @@ app.on('ready', () => {
 
 	page.on('dom-ready', () => {
 		page.insertCSS(fs.readFileSync(path.join(__dirname, 'browser.css'), 'utf8'));
+		page.insertCSS(fs.readFileSync(path.join(__dirname, 'darkMode.css'), 'utf8'));
```
Comment (sindresorhus) [inline index.js]: darkMode => dark-mode

File: darkMode.css [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	color: rgba(255, 255, 255, 0.9);
+}
+
+/* DM Send button */
+html.darkMode ._1LQ_VFHl {
+	border-color: #0084ff;
+}
+
+/* Trending item title */
+html.darkMode ._2WpoiLFQ {
```
Comment (sindresorhus) [inline darkMode.css]: `html.darkMode` => `html.dark-mode`

**Comment by danhp** [comment]
Done

File: browser.js [hunk 3]
```diff
@@ -192,13 +193,37 @@ function registerShortcuts(username) {
 	// -- //
 }
 
+function setDarkMode() {
+	document.documentElement.classList.toggle('dark-mode', storage.get('darkMode'));
+}
+
+function toggleDarkMode() {
```
Comment (sindresorhus) [inline browser.js]: Just put this inline in the event. It's not used anywhere else.

File: menu.js [hunk 4]
```diff
@@ -19,6 +19,16 @@ function sendAction(action, data) {
 
 const viewSubmenu = [
 	{
+		label: 'Toggle Dark Mode',
```
Comment (sindresorhus) [inline menu.js]: In Caprine we have this in the About menu. Really not sure what's the best location.

File: browser.js [hunk 5]
```diff
@@ -192,13 +193,33 @@ function registerShortcuts(username) {
...
+	});
+}
+
 function init() {
 	// hide navbar profile link
 	$('header a[href$="/account"]').parentNode.style.display = 'none';
 
 	const state = JSON.parse($('.___iso-state___').dataset.state).initialState;
 	const username = state.settings.data.screen_name;
+	if (process.platform === 'darwin') {
```
Comment (sindresorhus) [inline browser.js:220]: add an empty line above the `if` statement.

File: menu.js (hunk 4 above)
Comment (danhp) [inline menu.js]: I think we should move it so About for consistency. It works nice in Caprine as it's next to the settings shortcut.

File: browser.js [hunk 6]
```diff
@@ -192,13 +193,33 @@ function registerShortcuts(username) {
...
+		storage.set('darkMode', osxAppearance.isDarkMode());
+	}
 
+	setDarkMode();
```
Comment (sindresorhus) [inline browser.js:225]: add a comment above this:

```
// activate Dark Mode if it was set before quitting
```

File: browser.js (hunk 5 above)
Comment (sindresorhus) [inline browser.js:220]: also add comment above.

```
// link the theme if it was changed while the app was closed
```

File: menu.js (hunk 4 above)
Comment (sindresorhus) [inline menu.js]: 👍

**Comment by danhp** [comment]
And now I think it's done.

**Comment by sindresorhus** [comment]
- Tweet text need to be white. It's currently gray.
- The headings need to be a bit brighter, e.g. `Home`, `Messages`.
- The DM separators in the Messages tab are too bright.
- The chat bubble in DM conversations needs to be styled, the top & bottom separator too.
- In Search, the live dropdown when searching needs to be styled.
- In the Profile, the name and description needs to be brighter.
- The bottom border when composing a new tweet needs to be darker.

**Comment by paulmolluzzo** [comment]
So is my PR just not going to be used then? It looks like it was copied over manually and had stuff appended.

**Comment by sindresorhus** [comment]
@paulmolluzzo Sorry... I was going to merge yours today, but this one is more in line with how I want it. Mainly because it's a direct port of the Dark Mode in [Caprine](https://github.com/sindresorhus/caprine), which we've iterated on a lot there.

**Comment by paulmolluzzo** [comment]
@sindresorhus no worries

**Comment by danhp** [comment]
@paulmolluzzo Sorry for taking over like this... I'll try not to do this again.

**Comment by paulmolluzzo** [comment]
Looks good to me. 👍 

My only comment, which isn't a deal breaker, is the BG color seems a little bright. What if we change the colors to use the [official Twitter palette](https://about.twitter.com/company/brand-assets)? If all we change is the background that would make it `#292f33`.

**Comment by sindresorhus** [comment]
Landed! Thank you @danhp :) And @paulmolluzzo for starting it.

**Comment by sindresorhus** [comment]
> My only comment, which isn't a deal breaker, is the BG color seems a little bright. What if we change the colors to use the official Twitter palette? If all we change is the background that would make it #292f33.

It seems like `#292f33` is actually brighter. I also prefer the dark bluishness of the current scheme.

## #292f33

screen shot 2016-05-12 at 13 16 25

## Current background

screen shot 2016-05-12 at 13 15 45

## 

For reference, here's the dark mode in the official Twitter app:

screen shot 2016-05-12 at 13 02 29

Which is `#192632`, while ours is `#192633`. So very close.

**Comment by sindresorhus** [comment]
Should we maybe use blue links like the official Twitter app?

**Comment by sindresorhus** [comment]
I also notice the official app uses a background instead of just border for inline referenced tweets. Should we too?

**Comment by paulmolluzzo** [comment]
@sindresorhus Cool, looks like it matches the official app. :+1: for changing link color and BG on embedded tweets.

**Comment by sindresorhus** [comment]
@paulmolluzzo @danhp Hey guys. Was wondering if you'd be interested in officially joining this project? You've both already done some great contributions and I value your opinions. No worries if not though :)

**Comment by paulmolluzzo** [comment]
@sindresorhus Sure thing! Thanks! Actually, I thought you added me already...

**Comment by danhp** [comment]
@sindresorhus Yup! Wouldn't mind helping out.

**Comment by sindresorhus** [comment]
Awesome! Glad to have you onboard :)

**Comment by sindresorhus** [comment]
> Actually, I thought you added me already...

Yeah sorry, I just never got around to actually invite you until now ;)

**Pull request closed without merging**